navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -295,10 +327,26 @@ public void startNavigation(@NonNull DirectionsRoute directionsRoute) {
     this.directionsRoute = directionsRoute;
     Timber.d("MapboxNavigation startNavigation called.");
     if (!isBound) {
+      // Navigation sessions initially starting
+      sessionState = SessionState.builder()
+        .originalDirectionRoute(directionsRoute)
+        .currentDirectionRoute(directionsRoute)
+        .sessionIdentifier(TelemetryUtils.buildUUID())
```

Comment:
This constructs a new UUID every time, rather than a UUID that represents the current build, right?

Reply:
Nah, the `if(!isBound)` is only true when the user begins a navigation session. The id does get recreated if the user cancels and then starts the session again. Is this what we want?

Reply:
> The id does get recreated if the user cancels and then starts the session again. Is this what we want?

Yup. That's the correct behavior.